addons/account/models/partner.py

```diff
@@ -235,7 +239,7 @@ def _get_fiscal_position(self, partner, delivery=None):
             return manual_fiscal_position
 
         # First search only matching VAT positions
-        vat_required = bool(partner.vat)
+        vat_required = self._get_vat_required(partner, delivery, company)
```

Comment:
I'm not sure I understand this, but I'm sure that some tests covering the 3 cases of your commit message would help me :smile: